lib/usb/usb_msc.c

```diff
@@ -476,6 +489,25 @@ static void scsi_inquiry(usbd_mass_storage *ms,
 	}
 }
 
+static void scsi_read_format_capacities(usbd_mass_storage *ms, struct usb_msc_trans *trans, enum trans_event event)
+{
+	if (EVENT_CBW_VALID == event) {
+		trans->msd_buf[3] = 0x08;
+		trans->msd_buf[4] = ms->block_count >> 24;
+		trans->msd_buf[5] = 0xff & (ms->block_count >> 16);
+		trans->msd_buf[6] = 0xff & (ms->block_count >> 8);
+		trans->msd_buf[7] = 0xff & ms->block_count;
+
+		trans->msd_buf[8] =  0x02;
+		trans->msd_buf[9] = 0;
+		trans->msd_buf[10] = 0x02;
+		trans->msd_buf[11] = 0;
+		trans->bytes_to_write = 9;
+		set_sbc_status_good(ms);
+	}
```

Comment:
else set status bad? what other events are not handled here?

Reply:
>else set status bad? 

"in beginning everything was bad" =)

Karl, this is not my code change. Original of patch there https://github.com/thirdpin/libopencm3
I do not have time now to work with it.